test/e2e/main_test.go

```diff
@@ -194,130 +229,219 @@ func TestInstallMeteringAndReportingProducesData(t *testing.T) {
-	// be run in a goroutine - so check if the cfg object is nil to avoid a panic
-	if rf != nil {
-		// note: we need to run the testReportingProducesData as a group in order
-		// to avoid preemptively running the defer closure early and uninstalling
-		// the metering resources during the middle of the test execution
-		// for more information, see:
-		// https://github.com/golang/go/issues/22993
-		// https://github.com/golang/go/issues/31651
-		t.Run("testReportingProducesData", func(t *testing.T) {
-			testReportingProducesData(t, rf)
-		})
-	}
-}
+		df.Logger.Infof("Storing logs for the %s testcase at %s", testFunction.Name, testFunctionOutputDir)
 
-func testReportingProducesData(t *testing.T, testReportingFramework *reportingframework.ReportingFramework) {
-	// cron schedule to run every minute
-	cronSchedule := &metering.ReportSchedule{
-		Period: metering.ReportPeriodCron,
-		Cron: &metering.ReportScheduleCron{
-			Expression: fmt.Sprintf("*/1 * * * *"),
-		},
-	}
+		deployerCtx, err := df.NewDeployerCtx(
+			testFuncNamespace,
+			meteringOperatorImageRepo,
+			meteringOperatorImageTag,
+			reportingOperatorImageRepo,
+			reportingOperatorImageTag,
+			testMeteringConfigSpec,
+			testFunctionOutputDir,
+			testTargetPods,
+		)
+		require.NoError(t, err, "creating a new deployer context should produce no error")
 
-	queries := []struct {
-		queryName   string
-		skip        bool
-		nonParallel bool
-	}{
-		{queryName: "namespace-cpu-request"},
-		{queryName: "namespace-cpu-usage"},
-		{queryName: "namespace-memory-request"},
-		{queryName: "namespace-persistentvolumeclaim-request"},
-		{queryName: "namespace-persistentvolumeclaim-usage"},
-		{queryName: "namespace-memory-usage"},
-		{queryName: "persistentvolumeclaim-usage"},
-		{queryName: "persistentvolumeclaim-capacity"},
-		{queryName: "persistentvolumeclaim-request"},
-		{queryName: "pod-cpu-request"},
-		{queryName: "pod-cpu-usage"},
-		{queryName: "pod-memory-request"},
-		{queryName: "pod-memory-usage"},
-		{queryName: "node-cpu-utilization"},
-		{queryName: "node-memory-utilization"},
-		{queryName: "cluster-persistentvolumeclaim-request"},
-		{queryName: "cluster-cpu-capacity"},
-		{queryName: "cluster-memory-capacity"},
-		{queryName: "cluster-cpu-usage"},
-		{queryName: "cluster-memory-usage"},
-		{queryName: "cluster-cpu-utilization"},
-		{queryName: "cluster-memory-utilization"},
-		{queryName: "namespace-memory-utilization"},
-		{queryName: "namespace-cpu-utilization"},
-		{queryName: "pod-cpu-request-aws", skip: !runAWSBillingTests, nonParallel: true},
-		{queryName: "pod-memory-request-aws", skip: !runAWSBillingTests, nonParallel: true},
-		{queryName: "aws-ec2-cluster-cost", skip: !runAWSBillingTests, nonParallel: true},
-	}
+		rf, canSafelyRunTest, err := deployerCtx.Setup(testExpectErr)
+		if testExpectErr {
+			assert.NotNil(t, err, "expected the install to fail but it didnt")
+			if err != nil {
+				assert.Contains(t, err.Error(), testExpectErrMsg)
```

Comment:
This could be problematic as `err` is just an interface so null checking might not be the correct implementation and would result in a panic.

If we changed the `fmt.Errorf(...)` in the deploy package to `errors.Wrap(err, "...")` we might be able to use `assert.EqualErrors(t, errors.Cause(err), testExpectErrMsg)` to get the last error message in the stack trace, which is safer and eliminates the need to also check if `assert.NotNil(t, err, ...)`.